packages/flutter_tools/test/general.shard/base/logger_test.dart

```diff
@@ -1079,3 +1207,47 @@ class FakeStopwatchFactory implements StopwatchFactory {
     return stopwatch ?? FakeStopwatch();
   }
 }
+
+/// A fake [Logger] that throws the [Invocation] for any method call.
+class FakeLogger implements Logger {
+  @override
+  dynamic noSuchMethod(Invocation invocation) => throw invocation;
+}
+
+/// Returns whether two [Invocation]s should be considered equal.
+bool _invocationsEqual(Invocation invocation1, Invocation invocation2) {
```

Comment:
This won't give you any information about what is different between the two calls, beyond examining the string interpolated output. There are a few patterns you could use to make this easier to update.

* Use a TypeMatcher, for example: https://github.com/flutter/flutter/blob/7f2ca5e532b4052bd1e00d466493f9b583a7ca23/packages/flutter_tools/test/general.shard/runner/local_engine_test.dart#L124

* Use expect for the matching, which will throw an exception on the exact malformed bit https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/src/fake_process_manager.dart#L104

Reply:
Yeah, I wasn't happy about not having detailed failure messages.

I was not familiar with `TypeMatcher` before.  Thanks!  I switched to using that and forced some failures.  They still don't have great readability, but they should be adequate to identify which part of the `Invocation` failed to match.